File: javascript/resistor-color-trio/resistor-color-trio.js
export class ResistorColorTrio {

    constructor(bands) {
        let colorCodes = ["black", "brown" ,"red", "orange",
                            "yellow", "green", "blue", "violet",
                            "grey", "white"];

        let value0 = colorCodes.indexOf(bands[0]);
        let value1 = colorCodes.indexOf(bands[1]);
        let zeros = colorCodes.indexOf(bands[2]);
        
        if (value0 === -1 || value1 == -1 || zeros === -1) {
            throw new Error('invalid color'); 
        }
        
        let totalValue = (value0 * 10 + value1) * Math.pow(10, zeros);
        let usedUnit = totalValue < 1000 ? 'ohms' : 'kiloohms';
        totalValue = totalValue > 1000 ? totalValue / 1000 : totalValue;

        this.label = `Resistor value: ${totalValue} ${usedUnit}`;

    }

    label() {
        return this.label;
    }
}

File: java/resistor-color-trio/src/main/java/ResistorColorTrio.java
import java.util.Map;

class ResistorColorTrio {
    String label(String[] colors) {
        Map<String, Integer> bandColors = Map.of(
                "black", 0, "brown", 1, "red", 2, "orange", 3, "yellow", 4, 
                "green", 5, "blue", 6, "violet", 7, "grey", 8, "white", 9);

        long resistance = ((bandColors.get(colors[0]) * 10) + bandColors.get(colors[1]))
                            * (long)Math.pow(10, bandColors.get(colors[2]));

        return resistance >= 1_000_000_000 ? (resistance / 1_000_000_000) + " gigaohms"
                        : resistance >= 1_000_000 ? (resistance / 1_000_000) + " megaohms"
                        : resistance >= 1_000 ? (resistance / 1_000) + " kiloohms" 
                        : resistance + " ohms";

    }
}
